.github/composites/docker-build-push/action.yaml

```diff
@@ -0,0 +1,115 @@
+name: 'Build and Publish Docker Image'
+description: 'Build and Publish Docker Image'
+inputs:
+  repository:
+    description: 'Docker hub repository'
+    required: true
+    default: 'cloudcustodian'
+  name:
+    description: 'image name, should match dockerfile name'
+    required: true
+  push:
+    description: 'Push image: true/false'
+    required: true
+    default: false
+  poetry_version:
+    description: "Poetry Version to use"
+    default: "1.1.14"
+  trivy_version:
+    description: "trivy version to use"
+    default: "0.5.4"
+  platforms:
+    description: "Platforms to build, e.g. linux/arm64,linux/amd64"
+    default: "linux/amd64"
+runs:
+  using: "composite"
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+
+    # until we get read only credentials, we'll only login if we're pushing
+    # note that we can hit rate limits on pulling though
+    - name: Login to Docker Hub
+      if: "${{ inputs.push == 'true' }}"
+      uses: docker/login-action@v1
+      with:
+        username: ${{ env.HUB_USER }}
+        password: ${{ env.HUB_TOKEN }}
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          ${{ inputs.repository }}/${{ inputs.name }}
+        tags: |
+          type=semver,pattern={raw}
+          type=raw,value=latest
```

Comment:
we want to set the value here conditionally if its a push event to trunk or release == latest else dev

Reply:
i thought only releases would result in a latest tag?

Reply:
Your right only releases for latest tag, everything else gets dev

Reply:
ack